src/CodeGenerator/src/Generator/InputGenerator.php

```diff
@@ -154,8 +154,25 @@ public function generate(Operation $operation): ClassName
                         'NAME' => $member->getName(),
                         'CLASS' => $memberClassName->getName(),
                     ]);
-                } elseif ($mapValueShape instanceof ListShape || $mapValueShape instanceof MapShape) {
-                    throw new \RuntimeException('Recursive ListShape are not yet implemented');
+                } elseif ($mapValueShape instanceof ListShape) {
+                    $listMember = $mapValueShape->getMember();
+                    $listMemberShape = $listMember->getShape();
+                    if (!$listMemberShape instanceof StructureShape) {
+                        throw new \RuntimeException('Recursive ListShape with non StructureShape member is not implemented.');
+                    }
+                    $memberClassName = $this->objectGenerator->generate($listMemberShape);
+                    $constructorBody .= strtr('
+                        $this->NAME = [];
+                        foreach ($input["NAME"] ?? [] as $key => $item) {
```

Comment:
are you sure about $key in a list ?

Reply:
Yes. In `BatchWriteItemInput` they specify the database table. 

https://github.com/async-aws/aws/pull/679/files#diff-c63388a34e43ac3b149f15b4b3c58cdcR20

Reply:
oh nvm... `foreach ($input["NAME"] ?? [] as $key => $item) {` is iterating over the Map. 
The iteration over the list is done next line with array_map.

Make sens